packages/nx-monorepo/src/components/nx-workspace.ts

```diff
@@ -228,13 +245,35 @@ export class NxWorkspace extends Component {
     );
   }
 
-  /** @internal */
-  protected _recursivelyApplyToProject(
+  /**
+   * Recursively apply a function a project and subprojects.
+   * @param project Base project to apply function and recursively to subprojects
+   * @param fn The function to apply to project
+   * @param predicate Predicate to match projects to apply function to
+   */
+  public recursivelyApplyToProject(
     project: Project,
-    fn: (project: Project) => void
+    fn: ProjectCallback,
+    predicate?: ProjectPredicateCallback
+  ): void {
+    (!predicate || predicate(project)) && fn(project);
+    project.subprojects.forEach((p) =>
```

Comment:
could this be: `recursivelyApplyToProjects(project.subprojects, fn, predicate);`